python/lsst/sims/photUtils/BandpassDict.py

```diff
@@ -0,0 +1,331 @@
+import copy
+import numpy
+from collections import OrderedDict
+from lsst.sims.photUtils import Bandpass, Sed
+
+__all__ = ["BandpassDict"]
+
+class BandpassDict(object):
+    """
+    This class will wrap an OrderedDict of Bandpass instantiations.
+
+    Upon instantiation, this class's constructor will resample
+    the input Bandpasses to be on the same wavelength grid (defined
+    by the first input Bandpass).  The constructor will then calculate
+    the 2-D phiArray for quick calculation of magnitudes in all
+    Bandpasses simultaneously (see the member method calcMagListFromSed).
+    """
+
+    def __init__(self, bandpassList, bandpassNameList):
+        """
+        @param [in] bandpassList is a list of Bandpass instantiations
+
+        @param [in] bandpassNameList is a list of tags to be associated
+        with those Bandpasses
+        """
+        self._bandpassDict = OrderedDict()
+        for bandpassName, bandpass in zip(bandpassNameList, bandpassList):
+
+            if bandpassName in self._bandpassDict:
+                raise RuntimeError("The bandpass %s occurs twice in your input " % bandpassName \
+                                   + "to BandpassDict")
+
+            self._bandpassDict[bandpassName] = copy.deepcopy(bandpass)
+
+        dummySed = Sed()
+        self._phiArray, self._wavelenStep = dummySed.setupPhiArray(self._bandpassDict.values())
```

Comment:
I am wary of touching Bandpass and Sed because the systems engineering teams use that code in ways that I am totally unaware of (i.e. I am unable to maintain it for them, should we choose to change the API).

setuUpPhiArray sets up one 2-D numpy array that contains the phi information for all of the involved bandpasses.  manyMagCalc operates on that one, monolithic phi array.  It achieves greater speed than calling calcMag on each bandpass because it assumes everything has already been sampled to the same wavelength grid.  That is also why the BandpassDict samples every bandpass to the same wavelength grid.

I envisioned the methods loadBandpassesFromFiles and loadTotalBandpassesFromFiles as the way to fulfill the "we need a method to construct bandpasses from a list of files" request.  I prefer to leave them outside of the BandpassDict class, because, if they were a part of BandpassDict, we would have to rewrite the BandpassDict constructor to do nothing (since it would not have throughput information until loadBandpassesFromFiles was called), which I am not comfortable with (though this is mostly a personal aesthetic choice).